packages/components/src/internal/solid-element.ts

```diff
@@ -34,6 +34,40 @@ export default class SolidElement extends LitElement {
 
     return event;
   }
+
+  /* eslint-disable */
+  static version = '';
+  /* eslint-enable */
+
+  static define(name: string, elementConstructor = this, options: ElementDefinitionOptions = {}) {
+    const currentlyRegisteredConstructor = customElements.get(name) as CustomElementConstructor | typeof SolidElement;
+
+    if (!currentlyRegisteredConstructor) {
+      customElements.define(name, class extends elementConstructor {} as unknown as CustomElementConstructor, options);
+      return;
+    }
+
+    let newVersion = ' (unknown version)';
+    let existingVersion = newVersion;
+
+    if ('version' in elementConstructor && elementConstructor.version) {
+      newVersion = ' v' + elementConstructor.version;
+    }
+
+    if ('version' in currentlyRegisteredConstructor && currentlyRegisteredConstructor.version) {
+      existingVersion = ' v' + currentlyRegisteredConstructor.version;
+    }
+
+    // Need to make sure we're not working with null or empty strings before doing version comparisons.
+    if (newVersion && existingVersion && newVersion === existingVersion) {
+      // If versions match, we don't need to warn anyone. Carry on.
+      return;
+    }
+
+    console.warn(
+      `Attempted to register <${name}>${newVersion}, but <${name}>${existingVersion} has already been registered.`
```

Comment:
Why is this here?

Reply:
This is was one of the dependencies I added for the first version of carousel. But I see it is no longer needed after we switched the version.   I can go ahead and remove it now.